app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -0,0 +1,90 @@
+package com.example.nicocalculator.mvp.presenter
+
+import android.widget.Button
+import com.example.nicocalculator.Key
+import com.example.nicocalculator.mvp.model.CalculatorModel
+import com.example.nicocalculator.mvp.view.CalculatorView
+
+class CalculatorPresenter(private val model: CalculatorModel, private val view: CalculatorView) {
+
+    companion object {
+        private const val VALUE_EMPTY_TEXT = ""
+        private const val VALUE_ZERO = 0
+        private const val DECIMAL_SEPARATOR = '.'
+        private const val VALUE_MIN_CAR = 48
+        private const val VALUE_MAX_CHAR = 57
+        private const val ADD_OPERATOR = "+"
+        private const val SUBTRACT_OPERATOR = "-"
+        private const val MULTIPLY_OPERATOR = "*"
+        private const val DIVIDE_OPERATOR = "/"
+    }
+
+    fun onDeleteButtonPressed() {
+        var textOperations = model.getTextOperations()
+        textOperations =
+            if (textOperations.isNotEmpty()) textOperations.subSequence(0 until textOperations.length - 1) as String
+            else VALUE_EMPTY_TEXT
+        model.setTextOperations(textOperations)
+        view.setTextOperations(textOperations)
+    }
+
+    fun onEraseButtonPressed() {
+        model.setTextOperations(VALUE_EMPTY_TEXT)
+        view.setTextOperations(VALUE_EMPTY_TEXT)
+    }
+
+    fun onKeyButtonPressed(button: Button) {
+        var textOperations = model.getTextOperations()
+        if (textOperations == CalculatorModel.OPERATIONS_DEFAULT_TEXT) {
+            textOperations = VALUE_EMPTY_TEXT
+        }
+        textOperations += button.text
+        model.setTextOperations(textOperations)
+        view.setTextOperations(textOperations)
+    }
+
+    fun onEqualButtonPressed() {
+        var textResult = getResults(getItems(model.getTextOperations()))
+        model.setTextResult(textResult)
+        view.setTextResult(textResult)
+    }
+
+    private fun getResults(itemList: List<Key>): String {
```

Comment:
You could move this to the model: if in the future a change is made in the way the results are handled, the presenter will  remain the same, ignoring that change, and you will only need to modify the model.

Reply:
Done